code/obj/machinery/manufacturer.dm

```diff
@@ -261,330 +265,221 @@ TYPEINFO(/obj/machinery/manufacturer)
 			src.take_damage(damage)
 
 	power_change()
-		if(status & BROKEN)
+		if(src.is_broken())
 			src.build_icon()
 		else
-			if(powered() && src.dismantle_stage < 3)
+			if(src.powered() && src.dismantle_stage < 3)
 				status &= ~NOPOWER
 				src.build_icon()
 			else
 				SPAWN(rand(0, 15))
 					status |= NOPOWER
 					src.build_icon()
 
+	// Overriden to not disable if no power, wire maintenence to restore power is on the GUI which creates catch-22 situation
+	broken_state_topic(mob/user)
+		. = user.shared_ui_interaction(src)
+		if (src.is_broken())
+			return min(., UI_CLOSE)
+		else if (requires_power && status & (NOPOWER | POWEROFF))
+			return min(., UI_INTERACTIVE)
+		else if (status & MAINT)
+			return min(., UI_UPDATE)
+
+	ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if(!ui)
+			ui = new(user, src, "Manufacturer")
+			ui.open()
+
+	ui_data(mob/user)
+		// When we update the UI, we must regenerate the blueprint data if the blueprints known to us has changed since last time
+		// No need to do this if we're depowered/broken though
+		if (should_update_static && !src.is_disabled())
+			should_update_static = FALSE
+			src.update_static_data(user)
```

Comment:
if it doesnt thats cool, ill remove it and see what happens

Reply:
It's a required param sadly, I kind of hate how this works because it means you have to iterate all the users of the UI to fully invalidate the cache when something changes but 🤷